src/Feed/ProductWalker.php

```diff
@@ -62,15 +70,67 @@ class ProductWalker {
 	 * @param ProductMapperInterface $mapper The product mapper.
 	 * @param FeedValidatorInterface $validator The feed validator.
 	 * @param FeedInterface          $feed The feed.
+	 * @param array                  $query_args The query arguments.
 	 */
-	public function __construct(
+	private function __construct(
 		ProductMapperInterface $mapper,
 		FeedValidatorInterface $validator,
-		FeedInterface $feed
+		FeedInterface $feed,
+		array $query_args
 	) {
-		$this->mapper    = $mapper;
-		$this->validator = $validator;
-		$this->feed      = $feed;
+		$this->mapper     = $mapper;
+		$this->validator  = $validator;
+		$this->feed       = $feed;
+		$this->query_args = $query_args;
+	}
+
+	/**
+	 * Creates a new instance of the ProductWalker class based on an integration.
+	 *
+	 * The walker will mostly be set up based on the integration.
+	 * The feed is provided externally, as it might be based on the context (CLI, REST, Action Scheduler, etc.).
+	 *
+	 * @param IntegrationInterface $integration The integration.
+	 * @param FeedInterface        $feed        The feed.
+	 * @return self The ProductWalker instance.
+	 */
+	public static function from_integration(
+		IntegrationInterface $integration,
+		FeedInterface $feed
+	): self {
+		$query_args = array_merge(
+			[
+				'status' => [ 'publish' ],
+				'return' => 'objects',
+			],
+			$integration->get_product_feed_query_args()
+		);
+
+		/**
+		 * Allows the base arguments for querying products for product feeds to be changed.
```

Comment:
Just wondering what's the use case of the filter, as each integration can already provide the args. Is it perhaps for some use cases that only differ in the query args from an existing integration? Could it affect the query args for the core use cases?

Reply:
> what's the use case of the filter

For other extensions to be able to integrate with product feeds. It can definitely affect query args for core use cases, that is the point. It's up to those extensions not to mess it up, as it is with most filters.